tests/foreman/ui/test_reporttemplates.py

```diff
@@ -342,3 +398,71 @@ def test_negative_nonauthor_of_report_cant_download_it(session):
     :expectedresults: Report can't be downloaded. Error.
     :CaseImportance: High
     """
+
+
+@tier3
+def test_positive_gen_entitlements_reports_multiple_formats(session, content_setup, module_org):
+    """Generate reports using the Entitlements template in html, yaml, json, and csv format.
+
+        :id: b268663d-c213-4e59-8f81-61bec0838b1e
+
+
+        :setup: Installed Satellite with Organization, Activation key,
+                Content View, Content Host, and Subscriptions.
+
+        :steps:
+            1. Monitor -> Report Templates
+            2. Entitlements -> Generate
+            3. Click the dropdown to select output format
+            4. Submit
+
+        :expectedresults: reports are generated containing all the expected information
+                          regarding Entitlements for each output format.
+
+        :CaseImportance: High
+    """
+    with VirtualMachine(distro=DISTRO_RHEL7) as vm:
+        vm.install_katello_ca()
+        module_org, ak = content_setup
+        vm.register_contenthost(module_org.label, ak.name)
+        assert vm.subscribed
+        with session:
+            session.location.select('Default Location')
+            result_json = session.reporttemplate.generate(
+                "Entitlements", values={'output_format': 'JSON'}
+            )
+            with open(result_json) as json_file:
+                data_json = json.load(json_file)
+            assert any(entitlement['Name'] == vm.hostname for entitlement in data_json)
+            assert any(
+                entitlement['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
+                for entitlement in data_json
+            )
```

Comment:
repeated three times, can't it go in some common function here? e.g. 

```python
def assert_reports_file( file_data, entitlement):
        assert any(entitlement['Name'] == vm.hostname for entitlement in file_data)
        assert any(
                entitlement['Subscription Name'] == DEFAULT_SUBSCRIPTION_NAME
                for entitlement in file_data
            )
```

Reply:
I think it's fine to be more explicit within the tests, so abstracting these checks away isn't much of a priority.

Reply:
I thought the assertion was repeating almost 4 times, with the same line. If I would be writing the code I would not repeat that code placed in a single place. I felt while I was reviewing the code. I won't take more than 10 mins.